core/src/main/java/org/apache/james/core/quota/QuotaCurrentValue.java

```diff
@@ -0,0 +1,74 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.core.quota;
+
+import java.util.Objects;
+
+public class QuotaCurrentValue {
+
+    public static QuotaCurrentValue of(QuotaComponent quotaComponent, String identifier, QuotaType quotaType, long currentValue) {
+        return new QuotaCurrentValue(quotaComponent, identifier, quotaType, currentValue);
+    }
+
+    private final QuotaComponent quotaComponent;
+    private final String identifier;
+    private final QuotaType quotaType;
+    private final long currentValue;
```

Comment:
Optional<Long> ? Maybe there is no quota yet for a user that just got created  and didnt receive any message yet for example?

Reply:
I think initially it should be zero

Reply:
In fact, this value in in counter table. I think it should never be null

Reply:
Not necessary? What if it's not? As said above user just got created for example..?

Reply:
Copied from cassandra official web page:
"The counter type is used to define counter columns. A counter column is a column whose value is a 64-bit signed integer and on which 2 operations are supported: incrementing and decrementing (see the [UPDATE](https://cassandra.apache.org/doc/latest/cassandra/cql/dml.html#update-statement) statement for syntax). Note that the value of a counter cannot be set: a counter does not exist until first incremented/decremented, and that first increment/decrement is made as if the prior value was 0"

Reply:
`a counter does not exist until first incremented/decremented`

Which was the point... what if the counter does not exist yet for that user? Turning in circles here :)

Reply:
If the counter does not exist, the whole row in cassandra table also does not exist. Therefore, when converting from a row to a QuotaCurrentValue object, the field currentValue of QuotaCurrentValue object always has a value